### ISSUE TITLE
Semantic segmentation timing out on all inputs

### ISSUE THREAD

**Issue by JRegimbal** [Shared-Reality-Lab/IMAGE-server#543]
The semantic segmentation preprocessor is regularly not responding and timing out to inputs in production. Logs for each request follow what is below (timestamps given for clarity)
```
image-semantic-segmentation  | 2023-01-18T20:14:20.144991143Z [2023-01-18 20:14:20 +0000] [807] [DEBUG] POST /preprocessor
image-semantic-segmentation  | 2023-01-18T20:14:20.145278606Z DEBUG:root:Received request
image-semantic-segmentation  | 2023-01-18T20:14:20.165502569Z INFO:root:Schemas loaded
image-semantic-segmentation  | 2023-01-18T20:14:50.974534070Z [2023-01-18 20:14:50 +0000] [1] [CRITICAL] WORKER TIMEOUT (pid:807)
image-semantic-segmentation  | 2023-01-18T20:14:51.999076100Z [2023-01-18 20:14:51 +0000] [1] [WARNING] Worker with pid 807 was terminated due to signal 9
image-semantic-segmentation  | 2023-01-18T20:14:52.000474719Z [2023-01-18 20:14:52 +0000] [885] [INFO] Booting worker with pid: 885
```
Note that this is not the semantic segmentation process returning after 30 seconds, but gunicorn itself killing the process and restarting. When requests are received, there is no significant increase in resource usage - it appears that the process is hanging.

This will not always occur. It is unclear what conditions lead to this state at this time.

**Comment by jeffbl** [comment]
@JRegimbal Just to be sure, can you include link to a photo that definitely reproduced the issue?

**Comment by JRegimbal** [comment]
On the IMAGE homepage, https://image.a11y.mcgill.ca/pages/imgs/pexels-photo-5926382.jpeg and https://image.a11y.mcgill.ca/images/pexels-photo-3584998_mod.jpeg. Note that I've restarted the container already and these work, so this seems to be more complicated than "these photos do not work".

**Comment by jeffbl** [comment]
As discussed, @rohanakut please add sufficient debug outpu to the semseg code to be able to capture what is going on when this happens again.

**Comment by rohanakut** [comment]
@jeffbl I have opened a PR for the same

**Comment by jeffbl** [comment]
Happening right now, on multiple photos I've tried. This is log from the bear picture on this page:
https://image.a11y.mcgill.ca/pages/tutorial.html

```
atp-orchestrator                   | 2023-03-13T14:27:23.543363867Z Received request                                                                                            
atp-orchestrator                   | 2023-03-13T14:27:23.556141391Z Running preprocessors in parallel...                                           
atp-orchestrator                   | 2023-03-13T14:27:23.556167070Z Now on priority group 1                                                         
atp-orchestrator                   | 2023-03-13T14:27:23.556173933Z Sending to preprocessor "autour-preprocessor"                                   
atp-orchestrator                   | 2023-03-13T14:27:23.559386997Z Sending to preprocessor "content-categoriser"                                   
image-autour-preprocessor          | 2023-03-13T14:27:23.563065409Z [2023-03-13 14:27:23 +0000] [7] [DEBUG] POST /preprocessor                      
image-content-categoriser          | 2023-03-13T14:27:23.563163895Z [2023-03-13 14:27:23 +0000] [9] [DEBUG] POST /preprocessor                     
atp-orchestrator                   | 2023-03-13T14:27:23.803170619Z Now on priority group 2                                                        
atp-orchestrator                   | 2023-03-13T14:27:23.803196488Z Sending to preprocessor "graphic-tagger"                                       
image-graphic-tagger               | 2023-03-13T14:27:23.806389754Z [2023-03-13 14:27:23 +0000] [7] [DEBUG] POST /preprocessor
atp-orchestrator                   | 2023-03-13T14:27:24.281877712Z Now on priority group 3
atp-orchestrator                   | 2023-03-13T14:27:24.281895015Z Sending to preprocessor "object-detection"
atp-orchestrator                   | 2023-03-13T14:27:24.283828023Z Sending to preprocessor "semantic-segmentation"
image-semantic-segmentation        | 2023-03-13T14:27:24.286488123Z [2023-03-13 14:27:24 +0000] [3011] [DEBUG] POST /preprocessor
image-object-detection             | 2023-03-13T14:27:24.286476340Z [2023-03-13 14:27:24 +0000] [31] [DEBUG] POST /preprocessor
image-semantic-segmentation        | 2023-03-13T14:27:24.286765837Z DEBUG:root:Received request
image-semantic-segmentation        | 2023-03-13T14:27:24.308225326Z INFO:root:Schemas loaded                                                                                    image-semantic-segmentation        | 2023-03-13T14:27:24.826162304Z INFO:root:converting base64 to numpy array
image-semantic-segmentation        | 2023-03-13T14:27:24.831672503Z INFO:root:graphic oiginal dimension (534, 800, 3)                           
atp-orchestrator                   | 2023-03-13T14:27:39.293136610Z Error occured fetching from semantic-segmentation                   
atp-orchestrator                   | 2023-03-13T14:27:39.293165615Z AbortError: The user aborted a request.                                                      
atp-orchestrator                   | 2023-03-13T14:27:39.293169112Z     at abort (/usr/src/app/node_modules/node-fetch/lib/index.js:1448:16)
atp-orchestrator                   | 2023-03-13T14:27:39.293171596Z     at EventTarget.abortAndFinalize (/usr/src/app/node_modules/node-fetch/lib/index.js:1463:4)
atp-orchestrator                   | 2023-03-13T14:27:39.293174021Z     at [nodejs.internal.kHybridDispatch] (node:internal/event_target:758:20)
atp-orchestrator                   | 2023-03-13T14:27:39.293176295Z     at EventTarget.dispatchEvent (node:internal/event_target:700:26)
atp-orchestrator                   | 2023-03-13T14:27:39.293178509Z     at abortSignal (node:internal/abort_controller:314:10)
atp-orchestrator                   | 2023-03-13T14:27:39.293180713Z     at AbortController.abort (node:internal/abort_controller:332:5)
atp-orchestrator                   | 2023-03-13T14:27:39.293182928Z     at Timeout._onTimeout (/usr/src/app/dist/server.js:127:32)
atp-orchestrator                   | 2023-03-13T14:27:39.293185222Z     at listOnTimeout (node:internal/timers:568:17)
atp-orchestrator                   | 2023-03-13T14:27:39.293187476Z     at process.processTimers (node:internal/timers:511:7) {
atp-orchestrator                   | 2023-03-13T14:27:39.293189720Z   type: 'aborted'
atp-orchestrator                   | 2023-03-13T14:27:39.293191945Z }
atp-orchestrator                   | 2023-03-13T14:27:39.293328052Z Now on priority group 4
atp-orchestrator                   | 2023-03-13T14:27:39.293332710Z Sending to preprocessor "object-grouping"
```

@JRegimbal it also seems to be throwing an extended error in the orchestrator, which I think is new? Looks like it continues fine, so I'm assuming this is expected behavior now on timeout? If not, would be separate issue.

**Comment by jeffbl** [comment]
@jaydeepsingh25 once #647 is closed, please deprecate this preprocessor as non-working, and mark it clearly in the README.md so that people do not attempt to use it. This can then be closed as "won't fix"
